git/test/test_git.py

```diff
@@ -153,3 +153,23 @@ def test_env_vars_passed_to_git(self):
         editor = 'non_existant_editor'
         with mock.patch.dict('os.environ', {'GIT_EDITOR': editor}):
             assert self.git.var("GIT_EDITOR") == editor
+
+    def test_environment(self):
```

Comment:
I don't see how I could test the `sshkey` context manager without accessing an actual repo. Also, how could I check whether the environment is actually passed into the git process? The former would cover the latter. Do you have ideas?